python/lsst/daf/butler/core/fileTemplates.py

```diff
@@ -443,6 +457,14 @@ def format(self, ref):
             if field_name in ("run", "collection"):
                 usedRunOrCollection = True
 
+            # Check for request for additional information from the dataId
+            if "." in field_name:
+                primary, secondary = field_name.split(".")
+                if primary in extras:
+                    element = extras[primary]
```

Comment:
Probably better to call this local variable "record"; "element" would refer to the keys in this dict, but "primary" is also totally reasonable for that since it reflects where the key is coming from in the template.